pkg/router/template/router_test.go

```diff
@@ -1094,6 +1094,51 @@ func TestCalculateServiceWeights(t *testing.T) {
 			},
 			expectedWeights: map[ServiceUnitKey]int32{},
 		},
+		{
+			name:      "a single service with one endpoint with weight 0",
```

Comment:
*nit* Endpoints don't have weights, services do.

```suggestion
			name:      "a single service with weight 0 with multiple endpoints",
```